src/OpenTelemetry.Contrib.Instrumentation.Wcf/TelemetryDispatchMessageInspector.cs

```diff
@@ -141,6 +147,11 @@ public void BeforeSendReply(ref Message reply, object correlationState)
                 }
 
                 activity.Stop();
+
+                if (!(Propagators.DefaultTextMapPropagator is TraceContextPropagator))
```

Comment:
can you add about why we are doing this here?

Reply:
There are 2 things being added on this PR:

* Set Activity.Current if it is lost.
* Clear baggage when request is done.

Both of these IMO are standard things any root (incoming) instrumentation should do. In the main repo, Asp.Net & Asp.Net Core instrumentation already do these tasks.

* Set Activity.Current if it is lost.

This is a convenience thing. We are about to jump into user code (Enrich). Making sure Activity.Current is correct is just to make sure the state is good and proper and there are no surprises for anything downstream.

>what would happen to the original Current, if it was not null?

A bit lower we are going to call Activity.Stop(), which will reset Activity.Current anyway.

* Clear baggage when request is done.

This is to clean up any dictionaries captured in any execution contexts across all threads the request might have jumped through. This clears any refs so the GC can clean up anything added to baggage. Similar to what Activity.Stop does. If we don't do this, we are potentially causing more memory to end up in gen2.